### PR TITLE
added cancellation support to PTVS.

### PR DESCRIPTION
**Pull request by heejaechang**
currently, it is using a hack to add the support since ILanguageClient doesn't provide a way for us to inject our CancellationStrategy at the right point.

but except the very first a few min once folder is opened, it should work as expected.

### CODE REVIEW

**Comment by heejaechang** [comment]
@huguesv can you take a look?

File: Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageClient.cs [hunk 1]
```diff
@@ -209,5 +216,62 @@ public class PythonAnalysisSettings {
...
+                this._jsonRpc = jsonRpc ?? throw new ArgumentNullException(nameof(jsonRpc));
+
+                this._cancellationFolderPath = Path.Combine(Path.GetTempPath(), "python-languageserver-cancellation", this._folderName);
+
+                this._cancellationStrategy = this._jsonRpc.CancellationStrategy;
+                this._jsonRpc.Disconnected += OnDisconnected;
+
+                try {
+                    Directory.CreateDirectory(this._cancellationFolderPath);
+                } catch {
```
Comment (huguesv) [inline Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageClient.cs:238]: please change these catch all to use `IsCriticalException` extension method like it is done at https://github.com/microsoft/PTVS/blob/63992ffc2d5e8453df0af4896ad3128275ec38f8/Python/Product/PythonTools/PythonTools/PythonFeedbackDiagnosticFileProvider.cs#L74

File: Python/Product/PythonTools/PythonTools.csproj [hunk 2]
```diff
@@ -198,6 +198,9 @@
       <Private>True</Private>
       <IncludeInVSIX>True</IncludeInVSIX>
     </Reference>
+    <Reference Include="StreamJsonRpc">
```
Comment (huguesv) [inline Python/Product/PythonTools/PythonTools.csproj:201]: please search for StreamJsonRpc in `Build\Common.Build.CSharp.targets` 
you should be able to change `net46` there to `netstandard2.0` and undo this change, as well as undo your change in PreBuild.ps1

**Comment by huguesv** [comment]
@heejaechang the rest looks good, thanks for doing this!

File: Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageClient.cs (hunk 1 above)
Comment (heejaechang) [reply Python/Product/PythonTools/PythonTools/LanguageServerClient/PythonLanguageClient.cs:238]: thank you. made not catch critical exceptions.

File: Python/Product/PythonTools/PythonTools.csproj (hunk 2 above)
Comment (heejaechang) [reply Python/Product/PythonTools/PythonTools.csproj:201]: ah, thank you. I guess I missed this one when I searched for StreamJsonRpc string in the folder.

**Comment by heejaechang** [comment]
@huguesv addressed your feedback.

**Review by huguesv — APPROVED** [review]

**Comment by heejaechang** [comment]
@huguesv I can't check this in. I don't have permission.
